<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: A wrapper function to perform gene2pathway test.</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>

<table width="100%" summary="page for gene2pathway_test {seq2pathway}"><tr><td>gene2pathway_test {seq2pathway}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>
A wrapper function to perform gene2pathway test.
</h2>

<h3>Description</h3>

<p>The function includes two part, one runs the classical Fisher's exact test, the other runs novel gene2pathway test.
</p>


<h3>Usage</h3>

<pre>
gene2pathway_test(dat, DataBase="GOterm", FisherTest=TRUE, EmpiricalTest=FALSE, 
                 method=c("FAIME", "KS-rank", "cumulative-rank"), 
                 genome=c("hg38","hg19","mm10","mm9"), alpha=5, 
                 logCheck=FALSE, na.rm=FALSE, B=100, min_Intersect_Count=5)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>dat</code></td>
<td>

<p>A data frame of gene expression or a matrix of sequencing-derived gene-level measurements. 
The rows of dat correspond to genes, and the columns correspond to sample profile 
(eg. Chip-seq peak scores, somatic mutation p-values, RNS-seq or micro-array gene expression values). 
Note that the rows must be labeled by official gene symbol. The values contained in dat should be either finite or NA.
</p>
</td></tr>
<tr valign="top"><td><code>DataBase</code></td>
<td>

<p>A character string assigns an R GSA.genesets object to define gene-set. User can call GSA.read.gmt 
to load customized gene-sets with a .gmt format. If not specified, GO defined gene sets (BP,MF,CC) will be used.
</p>
</td></tr>
<tr valign="top"><td><code>FisherTest</code></td>
<td>

<p>A Boolean value. By default is TRUE to excute the function of the Fisher's exact test. Otherwise, 
only excutes the function of gene2pathway test.
</p>
</td></tr>
<tr valign="top"><td><code>EmpiricalTest</code></td>
<td>

<p>A Boolean value. By default is FALSE for multiple-sample dat. When true, gene2pathway_test 
calculates empirical p-values for gene-sets.
</p>
</td></tr>
<tr valign="top"><td><code>method</code></td>
<td>

<p>A character string determines the method to calculate the pathway scores. Currently, &quot;FAIME&quot; (default), 
&quot;KS-rank&quot;, and &quot;cumulative-rank&quot; are supported.
</p>
</td></tr>
<tr valign="top"><td><code>genome</code></td>
<td>

<p>A character specifies the genome type. Currently, choice of &quot;hg38&quot;, &quot;hg19&quot;, &quot;mm10&quot;, and &quot;mm9&quot; is supported.
</p>
</td></tr>
<tr valign="top"><td><code>alpha</code></td>
<td>

<p>A positive integer, 5 by default. This is a FAIME-specific parameter. A higher value puts more weights 
on the most highly-expressed ranks than the lower expressed ranks.
</p>
</td></tr>
<tr valign="top"><td><code>logCheck</code></td>
<td>

<p>A Boolean value. By default is FALSE. When true, the function takes the log-transformed values of gene 
if the maximum value of sample profile is larger than 20.
</p>
</td></tr>
<tr valign="top"><td><code>na.rm</code></td>
<td>

<p>A Boolean value indicates whether to keep missing values or not when method=&quot;FAIME&quot;. By default is FALSE.
</p>
</td></tr>
<tr valign="top"><td><code>B</code></td>
<td>

<p>A positive integer assigns the total number of random sampling trials to calculate the empirical pvalues. 
By default is 100.
</p>
</td></tr>
<tr valign="top"><td><code>min_Intersect_Count</code></td>
<td>

<p>A number decides the cutoff of the minimum number of intersected genes when reporting Fisher's exact tested results.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list or data frame. If the parameter &quot;FisherTest&quot; is true, the result is a list including both reports 
for Fisher's exact test and the gene2pathway test. Otherwise, only reports the gen2pathway tested results.
</p>


<h3>Author(s)</h3>

<p>Xinan Yang
</p>


<h3>Examples</h3>

<pre>
data(dat_chip)
data(MsigDB_C5,package="seq2pathway.data")  
#generate a demo GSA.genesets object
demoDB &lt;- MsigDB_C5
x=100
for(i in 1:3) demoDB[[i]]&lt;-MsigDB_C5[[i]][1:x]
 
res&lt;-gene2pathway_test(dat=head(dat_chip), DataBase=demoDB, 
		FisherTest=FALSE, EmpiricalTest=FALSE, 
        method="FAIME", genome="hg19", min_Intersect_Count=1)
# check ther result
names(res)
res[[1]]
res[[2]]
## Not run: 
res&lt;-gene2pathway_test(dat=head(dat_chip), DataBase="BP", 
		FisherTest=FALSE, EmpiricalTest=FALSE, 
        method="FAIME", genome="hg19", min_Intersect_Count=1)

## End(Not run)       
</pre>

<hr /><div style="text-align: center;">[Package <em>seq2pathway</em> version 1.14.0 <a href="00Index.html">Index</a>]</div>
</body></html>
