<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: R wrapped python function to map genomic regions on the...</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>

<table width="100%" summary="page for runseq2gene {seq2pathway}"><tr><td>runseq2gene {seq2pathway}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>
R wrapped python function to map genomic regions on the sequence-level to genes.
</h2>

<h3>Description</h3>

<p>Annotate genome regions of interest to either the nearest TSS or a broader range of neighboring genes.
</p>


<h3>Usage</h3>

<pre>
runseq2gene(inputfile, 
            search_radius=150000, promoter_radius=200, promoter_radius2=100, 
            genome=c("hg38","hg19","mm10","mm9"), adjacent=FALSE, SNP=FALSE, 
            PromoterStop=FALSE,NearestTwoDirection=TRUE,UTR3=FALSE)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>inputfile</code></td>
<td>

<p>An R object input file that records genomic region information (coordinates). 
The file format could be data frame defined as:  
</p>

<ol>
<li><p>column 1
the unique IDs of genomic regions of interest (peaks, mutations, or SNPs)

</p>
</li>
<li><p>column 2
the chromosome IDs (eg. chr5 or 5)

</p>
</li>
<li><p>column 3
the start of genomic regions

</p>
</li>
<li><p>column 4
the end of genomic regions (for SNP and point mutations, 
the difference of start and end is 1bp)

</p>
</li>
<li><p>column 5...
Other custom defined information (option)

</p>
</li></ol>

<p>Or, the input format should be RangedData object(from R package IRanges) with value column.
</p>

<ol>
<li><p>column 1: space
the chromosome IDs (eg. chr5 or 5)

</p>
</li>
<li><p>column 2: ranges
the ranges of genomic regions

</p>
</li>
<li><p>column 3: name
the unique IDs of genomic regions of interest (peaks, mutations, or SNPs)

</p>
</li>
<li><p>more columns:
Other custom defined information (optional)

</p>
</li></ol>

</td></tr>
<tr valign="top"><td><code>search_radius</code></td>
<td>

<p>A non-negative integer, with which the input genomic regions can be assigned 
not only to the matched or nearest gene, but also with all genes 
within a search radius for some genomic region type. 
This parameter works only when the parameter &quot;SNP&quot; is FALSE. Default is 150000. 
</p>
</td></tr>
<tr valign="top"><td><code>promoter_radius</code></td>
<td>

<p>A non-negative integer. Default is 200. Promoters are here defined 
as upstream regions of the transcription start sites (TSS). 
User can assign the promoter radius, a suggested value is between 200 to 2000. 
</p>
</td></tr>
<tr valign="top"><td><code>promoter_radius2</code></td>
<td>

<p>A non-negative integer. Default is 100. 
Promoters are here defined as downstream regions after the transcription start sites (TSS).
</p>
</td></tr>
<tr valign="top"><td><code>genome</code></td>
<td>

<p>A character specifies the genome type. Currently, 
choice of &quot;hg38&quot;, &quot;hg19&quot;, &quot;mm10&quot;, and &quot;mm9&quot; is supported.
</p>
</td></tr>
<tr valign="top"><td><code>adjacent</code></td>
<td>

<p>A Boolean. Default is FALSE to search all genes within the search_radius. 
Using &quot;TRUE&quot; to find the adjacent genes only and ignore the parameters &quot;SNP&quot; and &quot;search_radius&quot;.
</p>
</td></tr>
<tr valign="top"><td><code>SNP</code></td>
<td>

<p>A Boolean specifies the input object type. FALSE by default to keep on searching 
for intron and neighboring genes. Otherwise, runseq2gene stops searching 
when the input genomic region is residing on exon of a coding gene.
</p>
</td></tr>
<tr valign="top"><td><code>PromoterStop</code></td>
<td>

<p>A Boolean, &quot;FALSE&quot; by default to keep on searching neighboring genes using the parameter &quot;search_radius&quot;. 
Otherwise, runseq2gene stops searching neighboring genes. This parameter has function 
only if an input genomic region maps to promoter of coding gene(s).
</p>
</td></tr>
<tr valign="top"><td><code>NearestTwoDirection</code></td>
<td>

<p>A boolean, &quot;TRUE&quot; by default to output the closest left and closest right coding genes with directions. 
Otherwise, output only the nearest coding gene regardless of direction.
</p>
</td></tr>
<tr valign="top"><td><code>UTR3</code></td>
<td>

<p>A boolean, &quot;FALSE&quot; by defalt to calculate the distance from genes' 5UTR. Otherwsie, calculate the distance from genes' 3UTR. 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with multiple columns.
</p>
<table summary="R valueblock">
<tr valign="top"><td><code>Columns 1 to 4</code></td>
<td>
<p>The same as the first four columns in the input file. </p>
</td></tr>
<tr valign="top"><td><code>PeakLength</code></td>
<td>
<p>An integer gives the length of the input genomic region. 
It is the number of base pairs between the start and end of the region. </p>
</td></tr>
<tr valign="top"><td><code>PeakMtoStart_Overlap</code></td>
<td>

<p>An integer gives the distance from the TSS of mapped gene to the middle of genomic region. 
A negative value indicates that TSS of the mapped gene is at the right of the peak. 
Otherwise, PeakMtoStart_Overlap reports a numeric range showing the location 
of overlapped coordinates (exon, intron, CDS, or UTR). </p>
</td></tr>
<tr valign="top"><td><code>type</code></td>
<td>

<p>A character specifies the relationship between the genomic region and the mapped gene.
</p>

<ol>
<li><p>&quot;Exon&quot;
any part of a genomic region overlaps the exon region of the mapped gene

</p>
</li>
<li><p>&quot;Intron&quot;
any part of a genomic region overlaps an intron region of the mapped gene
  
</p>
</li>
<li><p>&quot;cds&quot;
any part of a genomic region overlaps the CDS region

</p>
</li>
<li><p>&quot;utr&quot;
any part of a genomic region overlaps a UTR region

</p>
</li>
<li><p>&quot;promoter&quot;
any part of a genomic region overlaps the promoter region of the mapped gene 
when an intergenic region of mapped gene covers the input genomic region

</p>
</li>
<li><p>&quot;promoter_internal&quot;
any part of a genomic region overlaps the promoter region of the mapped gene 
when an adjacent TTS region of mapped gene covers the input genomic region

</p>
</li>
<li><p>&quot;Nearest&quot;
the mapped gene is the nearest gene if the genomic region is located in an intergenic region

</p>
</li>
<li><p>&quot;L&quot; and &quot;R&quot;
show the relative location of mapped genes when the input genomic region resides within a bidirectional region

</p>
</li>
<li><p>&quot;Neighbor&quot;
any mapped gene within the search radius but belongs to none of the prior types

</p>
</li></ol>

</td></tr>
<tr valign="top"><td><code>BidirectionalRegion</code></td>
<td>

<p>A Boolean indicates whether or not the input genomic region is in bidirectional region. 
<cite>&quot;A 'bidirectional gene pair' refers to two adjacent genes coded on opposite strands, 
with their 5' UTRs oriented toward one another.&quot; (from wiki http://en.wikipedia.org/wiki/Promoter_(genetics) )</cite>. 
NA means the genomic region is at exon or intron region. </p>
</td></tr>
<tr valign="top"><td><code>Chr</code></td>
<td>
<p>An integer gives chromosome number of mapped gene. </p>
</td></tr>
<tr valign="top"><td><code>TSS</code></td>
<td>
<p>An integer indicates transcription start site of mapped gene regardless of strand.  </p>
</td></tr>
<tr valign="top"><td><code>TTS</code></td>
<td>
<p>An integer indicates transcription termination site of mapped gene regardless of strand. </p>
</td></tr>
<tr valign="top"><td><code>strand</code></td>
<td>
<p>A character indicates whether mapped gene is in forward (+) or reverse (-) direction on chromosome.</p>
</td></tr>
<tr valign="top"><td><code>gene_name</code></td>
<td>
<p>A character gives official gene symbol of mapped genes.   </p>
</td></tr>
<tr valign="top"><td><code>source</code></td>
<td>
<p>A character gives gene source (Ensembl classification) of mapped genes.  </p>
</td></tr>
<tr valign="top"><td><code>transID</code></td>
<td>
<p>A character gives Ensemble transcript ID of mapped genes. </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Bin Wang
</p>


<h3>References</h3>

<p>Lawrence M, Huber W, Pages H, Aboyoun P, Carlson M, Gentleman R, Morgan M and Carey V (2013) 
&quot;Software for Computing and Annotating Genomic Ranges.&quot;.
<em>PLoS Computational Biology</em>, <b>9</b>.
</p>


<h3>Examples</h3>

<pre>
    data(Chipseq_Peak_demo)
    res=runseq2gene(inputfile=Chipseq_Peak_demo)  
</pre>

<hr /><div style="text-align: center;">[Package <em>seq2pathway</em> version 1.14.0 <a href="00Index.html">Index</a>]</div>
</body></html>
