<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: An function to perform the runseq2pathway algorithm(s).</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>

<table width="100%" summary="page for runseq2pathway {seq2pathway}"><tr><td>runseq2pathway {seq2pathway}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>
An function to perform the runseq2pathway algorithm(s).
</h2>

<h3>Description</h3>

<p>A wrapper function to perform seq2gene and gene2pathway in series.
</p>


<h3>Usage</h3>

<pre>
runseq2pathway(inputfile,
               search_radius=150000, promoter_radius=200, promoter_radius2=100, 
               genome=c("hg38","hg19","mm10","mm9"), adjacent=FALSE, SNP= FALSE, 
               PromoterStop=FALSE, NearestTwoDirection=TRUE,UTR3=FALSE,
               DataBase=c("GOterm"), FAIMETest=FALSE, FisherTest=TRUE,
               collapsemethod=c("MaxMean","function","ME",
               "maxRowVariance","MinMean","absMinMean","absMaxMean","Average"), 
               alpha=5, logCheck=FALSE, B=100, na.rm=FALSE, min_Intersect_Count=5)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>inputfile</code></td>
<td>

<p>An R object input file that records genomic region information (coordinates). 
The file format could be data frame defined as:  
</p>

<ol>
<li><p>column 1
the unique IDs of genomic regions of interest (peaks, mutations, or SNPs)

</p>
</li>
<li><p>column 2
the chromosome IDs (eg. chr5 or 5)

</p>
</li>
<li><p>column 3
the start of genomic regions

</p>
</li>
<li><p>column 4
the end of genomic regions (for SNP and point mutations, the difference of start and end is 1bp)

</p>
</li>
<li><p>column 5...
Other custom defined information (option)

</p>
</li></ol>

<p>Or, the input format should be GRanges object(from R package GenomicRanges) with value column.
</p>

<ol>
<li><p>column 1: space
the chromosome IDs (eg. chr5 or 5)

</p>
</li>
<li><p>column 2: ranges
the ranges of genomic regions

</p>
</li>
<li><p>column 3: name
the unique IDs of genomic regions of interest (peaks, mutations, or SNPs)

</p>
</li>
<li><p>more columns:
Other custom defined information (optional)

</p>
</li></ol>

</td></tr>
<tr valign="top"><td><code>search_radius</code></td>
<td>

<p>A non-negative integer, with which the input genomic regions can be assigned not only 
to the matched or nearest gene, but also with all genes within a search radius for 
some genomic region type. This parameter works only when the parameter &quot;SNP&quot; is FALSE. Default is 150000. 
</p>
</td></tr>
<tr valign="top"><td><code>promoter_radius</code></td>
<td>

<p>A non-negative integer. Default is 200. Promoters are here defined as upstream regions of 
the transcription start sites (TSS). User can assign the promoter radius, a suggested value is between 200 to 2000. 
</p>
</td></tr>
<tr valign="top"><td><code>promoter_radius2</code></td>
<td>

<p>A non-negative integer. Default is 100. Promoters are here defined as downstream regions 
after the transcription start sites (TSS).
</p>
</td></tr>
<tr valign="top"><td><code>genome</code></td>
<td>

<p>A character specifies the genome type. Currently, choice of &quot;hg38&quot;, &quot;hg19&quot;, &quot;mm10&quot;, and &quot;mm9&quot; is supported.
</p>
</td></tr>
<tr valign="top"><td><code>adjacent</code></td>
<td>

<p>A Boolean. Default is FALSE to search all genes within the search_radius. Using &quot;TRUE&quot; to find 
the adjacent genes only and ignore the parameters &quot;SNP&quot; and &quot;search_radius&quot;.
</p>
</td></tr>
<tr valign="top"><td><code>SNP</code></td>
<td>

<p>A Boolean specifies the input object type. FALSE by default to keep on searching for intron and 
neighboring genes. Otherwise, runseq2gene stops searching when the input genomic region is residing on exon of a coding gene.
</p>
</td></tr>
<tr valign="top"><td><code>PromoterStop</code></td>
<td>

<p>A Boolean, &quot;FALSE&quot; by default to keep on searching neighboring genes using the parameter &quot;search_radius&quot;. 
Otherwise, runseq2gene stops searching neighboring genes. This parameter has function only if an input genomic region maps to promoter of coding gene(s).
</p>
</td></tr>
<tr valign="top"><td><code>NearestTwoDirection</code></td>
<td>

<p>A boolean, &quot;TRUE&quot; by default to output the closest left and closest right coding genes with directions. 
Otherwise, output only the nearest coding gene regardless of direction.
</p>
</td></tr>
<tr valign="top"><td><code>UTR3</code></td>
<td>

<p>A boolean, &quot;FALSE&quot; by defalt to calculate the distance from genes' 5UTR. Otherwsie, calculate the distance from genes' 3UTR.   
</p>
</td></tr>
<tr valign="top"><td><code>DataBase</code></td>
<td>

<p>A character string assigns an R GSA.genesets object to define gene-set. User can call GSA.read.gmt to 
load customized gene-sets with a .gmt format. 
If not specified, a character &quot;GOterm&quot; by default, three categories of GO-defined gene sets (BP,MF,CC) will be used. 
Alternatively, user can specify a category by the choice of &quot;BP&quot;,&quot;MF&quot;,&quot;CC&quot;.
</p>
</td></tr>
<tr valign="top"><td><code>FAIMETest</code></td>
<td>

<p>A boolean values. By default is FALSE. When true, executes function of gene2pathway test using the FAIME method, 
which only functions when the fifth column of input file exsists and is a vector of scores or values.
</p>
</td></tr>
<tr valign="top"><td><code>FisherTest</code></td>
<td>

<p>A Boolean value. By default is TRUE to excute the function of the Fisher's exact test. Otherwise, 
only excutes the function of gene2pathway test.
</p>
</td></tr>
<tr valign="top"><td><code>collapsemethod</code></td>
<td>

<p>A character for determining which method to use when call the function collapseRows in package WGCNA. 
The function &quot;collapsemethod&quot; uses this paramter to call the collapseRows() function in package &quot;WGCNA&quot;.
</p>
</td></tr>  
<tr valign="top"><td><code>alpha</code></td>
<td>

<p>A positive integer, 5 by default. This is a FAIME-specific parameter. A higher value puts more weights on 
the most highly-expressed ranks than the lower expressed ranks.
</p>
</td></tr>
<tr valign="top"><td><code>logCheck</code></td>
<td>

<p>A Boolean value. By default is FALSE. When true, the function takes the log-transformed values of gene 
if the maximum value of sample profile is larger than 20.
</p>
</td></tr>
<tr valign="top"><td><code>na.rm</code></td>
<td>

<p>A Boolean value indicates whether to keep missing values or not when method=&quot;FAIME&quot;. By default is FALSE.
</p>
</td></tr>
<tr valign="top"><td><code>B</code></td>
<td>

<p>A positive integer assigns the total number of random sampling trials to calculate the empirical pvalues. 
By default is 100.
</p>
</td></tr>
<tr valign="top"><td><code>min_Intersect_Count</code></td>
<td>

<p>A number decides the cutoff of the minimum number of intersected genes when reporting Fisher's exact tested results.
</p>
</td></tr>  
</table>


<h3>Value</h3>

<p>An R list of several data frames. The results of function seq2gene, Fisher's exact test and gene2pathway test results are included.
</p>


<h3>Author(s)</h3>

<p>Bin Wang, Xinan Yang
</p>


<h3>References</h3>

<p>Langfelder P, Horvath S (2008)
WGCNA: an R package for weighted correlation network analysis.
<em>BMC Bioinformatics</em>,  9:559.
</p>
<p>Miller JA, Cai C, Langfelder P, Geschwind DH, Kurian SM, Salomon DR, Horvath S (2011)
Strategies for aggregating gene expression data: The collapseRows R function.
<em>BMC Bioinformatics</em>, 12:322.
</p>
<p>Lawrence M, Huber W, Pages H, Aboyoun P, Carlson M, Gentleman R, Morgan M and Carey V (2013) 
&quot;Software for Computing and Annotating Genomic Ranges.&quot;.
<em>PLoS Computational Biology</em>, <b>9</b>.
</p>


<h3>Examples</h3>

<pre>
	data(Chipseq_Peak_demo)
	require(seq2pathway.data)
	data(MsigDB_C5, package="seq2pathway.data")
  #generate a demo GSA.genesets object
	demoDB &lt;- MsigDB_C5
	x=10
	for(i in 1:3) demoDB[[i]]&lt;-MsigDB_C5[[i]][1:x]
       res3=runseq2pathway(inputfile=Chipseq_Peak_demo,
		genome="hg19", search_radius=100, promoter_radius=50, promoter_radius2=0,
		FAIMETest=TRUE, FisherTest=FALSE,  
		DataBase=demoDB, min_Intersect_Count=1)	
	names(res3)
	res3[[1]]
  ## Not run: 
   # an example to use FET
	res=runseq2pathway(inputfile=Chipseq_Peak_demo,
		genome="hg19", search_radius=100, promoter_radius=50, promoter_radius2=0,
		DataBase=MsigDB_C5, NearestTwoDirection=FALSE,
		collapsemethod="Average", min_Intersect_Count=1) 
   # an example to use FAIME
	res2=runseq2pathway(inputfile=Chipseq_Peak_demo,
		genome="hg19", search_radius=100, promoter_radius=50, promoter_radius2=0,
		FAIMETest=TRUE, FisherTest=FALSE,  
		DataBase=MsigDB_C5, min_Intersect_Count=1)
	
## End(Not run)
</pre>

<hr /><div style="text-align: center;">[Package <em>seq2pathway</em> version 1.14.0 <a href="00Index.html">Index</a>]</div>
</body></html>
